src/ripple/app/paths/impl/StrandFlow.h

```diff
@@ -144,14 +144,21 @@ flow (
                         *sb, *afView, ofrsToRm, EitherAmount (*maxIn));
                     limitStepOut = r.second;
 
-                    if (strand[i]->isZero (r.second) ||
-                        get<TInAmt> (r.first) != *maxIn)
+                    if (strand[i]->isZero(r.second))
+                    {
+                        JLOG(j.trace()) << "First step found dry";
+                        return {tecPATH_DRY, std::move(ofrsToRm)};
+                    }
+                    if (get<TInAmt> (r.first) != *maxIn)
                     {
                         // Something is very wrong
                         // throwing out the sandbox can only increase liquidity
                         // yet the limiting is still limiting
-                        JLOG (j.fatal()) << "Re-executed limiting step failed";
-                        assert (0);
+                        JLOG(j.fatal())
+                            << "Re-executed limiting step failed. r.first: "
```

Comment:
nit: my only comment is around these debug messages. `r.first` and `r.second` aren't particularly useful names in the log messages.
 
If I was debugging given this message I'd have to copy/paste a sub-string of it and search the codebase for where it's logged to get more context vs. getting more context from the log message itself.